qiskit_optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -232,6 +218,12 @@ def _solve_internal(
             eigen_result = self._min_eigen_solver.compute_minimum_eigenvalue(operator)
             # analyze results
             raw_samples = None
+            if not hasattr(eigen_result, "eigenstate"):
+                raise QiskitOptimizationError(
+                    "MinimumEigenOptimizer does not support this minimum eigensolver "
+                    f"{type(self._min_eigen_solver)}. "
+                    "You can use qiskit_algorithms.SamplingMinimumEigensolver instead."
+                )
```

Comment:
Instead of early type check of SamplingMinimumEigensolver or NumPyMinimumEigensolver, I added a result check when solve is invoked.

Reply:
It will certainly result in a more friendly error message that is easier to understand. This is of course detected though after its run compute_minimum_eigenvalue which may take a lot of time/resources that ends up being wasted. An IDE via the typehinting should flag VQE (which I think we we most concerned about in 0.5 since in the past you could use this) as not being valid to be used for this algorithm. I think its ok - we can see how it goes - one suggestion I had made was to get the class name string from the solver instance and compare that up front to see if it was one of the primitive based VQE's we know about ie - that being just a string check does not require anything to be imported. But we could always add this later if users are still tripping up on VQE - since 0.5 has been out a while hopefully users are familiar enough now. And its quite clear now in the docs what is supported...

![image](https://github.com/qiskit-community/qiskit-optimization/assets/40241007/31148daa-070d-460a-9cac-07c9646a480e)